src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1213,8 +1213,10 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		// Set the new shell launch config
 		this._shellLaunchConfig = shell; // Must be done before calling _createProcess()
 
-		// Kill and clear up the process, making the process manager ready for a new process
-		this._processManager.dispose();
+		if (!this._shellLaunchConfig.attachPersistentTerminal) {
+			// Kill and clear up the process, making the process manager ready for a new process
+			this._processManager.dispose();
+		}
```

Comment:
We probably still need this as it's prepping for reusing the terminal

Reply:
doesn't look like I removed it?

Reply:
I mean this part:

```ts
if (!this._shellLaunchConfig.attachPersistentTerminal) {
```

Even if it was attached, I think we would still want to dispose if reuseTerminal is called.